includes/class-wp-job-manager-ui-notice.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Notice component for the frontend.
+ *
+ * @package wp-job-manager
+ * @since $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Notice component for the frontend.
+ *
+ * @since $$next-version$$
+ *
+ * @internal This API is still under development and subject to change.
+ * @access private
+ */
+class WP_Job_Manager_UI_Notice {
+
+	/**
+	 * The singleton instance of the class.
+	 *
+	 * @var self
+	 */
+	private static $instance = null;
+
+	/**
+	 * Singleton instance getter.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Notice element.
+	 *
+	 * @param array $options {
+	 *    Options for the notice.
+	 *
+	 * @type string $title Optional title.
+	 * @type string $icon Notice icon. Should be a supported icon name, or safe SVG code.
+	 * @type string $message Main notice message.
+	 * @type string $message_icon Alternative to icon, if it should be placed next to the message, not the title.
+	 * @type string $details Additional details.
+	 * @type array  $buttons Action links styled as buttons.
+	 * @type array  $links Action links.
+	 * @type array  $classes Additional classes for the notice.
+	 * }
+	 *
+	 * @return string HTML.
+	 */
+	public static function notice( $options ) {
```

Comment:
Why don't we pass each option as an argument? Plain arguments are easier to document. 

Also, they are easier to maintain as you need to track down how the options array is populated if you want to understand how a method is called.

Reply:
With this many arguments that's really hard to use and read. This is a pretty common pattern in WP too for complex methods, eg `get_posts`. But maybe in a few years we can use the [PHP 8 named arguments](https://stitcher.io/blog/php-8-named-arguments) :)

Reply:
> With this many arguments that's really hard to use and read

I see your point but encapsulating them in an array doesn't change anything, it just adds another layer. 

Named arguments is another reason not to have an array IMO as the way the function is now will never be able to benefit from it.

Reply:
It's what's possible right now for named arguments. This function has a lot of optional arguments, so calls could look like this:

`Notice:render( null, null, "Alert created", null, null, [[ 'url' => $manage_url, 'label' => 'Manage Alerts' ]])`

That's extremely hard to figure out, missing context for the what each argument is, and very easy to make mistakes with. 

There are some clean code literature on this: https://refactoring.guru/smells/long-parameter-list, https://dev.to/thinkster/code-smell-too-many-parameters-435e